src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,25 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position.
+     *
+     * @param idx Digit position we want to retrieve starting by 1 (not zero).
+     * @return The digit or -1 in case the position does not correspond with a digit.
+     */
     public int digit(int idx) {
-        Iterator i = items.iterator();
-        Item item = (Item) i.next();
-        while (idx > 0 && i.hasNext()) {
+        Iterator it = items.iterator();
+        int i = 1;
+        Item item = null;
+        while (i <= idx && it.hasNext()) {
+            item  = (Item) it.next();
             if (item instanceof IntegerItem) {
-                idx--;
+                i++;
+            } else {
+                return -1;
             }
-            i.next();
         }
-        return ((IntegerItem) item).value.intValue();
+        return ((idx <= 0) || (idx - i > 0)) ? -1 : ((IntegerItem) item).value.intValue();
```

Comment:
I would personally put the first check at the beginning of the method (you can avoid doing anything if that condition is not met). And I also see clearly (idx == i) than (idx - i > 0).

But it's probably something subjective so if you prefer this one, I am fine.